code/modules/food_and_drink/snacks.dm

```diff
@@ -1216,25 +1224,44 @@
 			user.show_text("The [tube] isn't full enough to add frosting.", "red")
 			return
 
-		if (src.style_step > length(src.styles))
+		if (src.style_step > 2) // only allow up to two frosting types on a single donut
 			user.show_text("You can't add anymore frosting.", "red")
 			return
 
+		var/frostingtype = null
+		frostingtype = input("Which frosting style would you like?", "Frosting Style", null) as null|anything in frostingstyle
+		if(frostingtype && (get_dist(src, user) <= 1))
+			switch(frostingtype)
+				if("icing")
+					frostingtype = "icing"
+				if("half and half icing")
+					frostingtype = "half"
+				if("dipped")
+					frostingtype = "dipped"
+				if("center fill")
+					frostingtype = "center"
+				if("zigzags")
+					frostingtype = "zigzags"
+				if("star")
+					frostingtype = "star"
+				if("heart")
+					frostingtype = "heart"
+			if(!src.GetOverlayImage(frostingstyle, src.layer + 0.2))
+				var/frostingstyle[src.style_step]
+				var/datum/color/average = tube.reagents.get_average_color()
+				var/image/frostingoverlay = new(src.icon, frostingtype)
+				frostingoverlay.color = average.to_rgba()
+				src.UpdateOverlays(frostingoverlay, frostingstyle)
```

Comment:
```suggestion
				src.UpdateOverlays(frostingoverlay, "frosting[src.style_step]")
```
Maybe this is what you intended or maybe not. But this makes it so the overlay slots will be named "frosting0" and "frosting1" which sounds good to me.